src/gui/mzroll/point.cpp

```diff
@@ -129,61 +129,61 @@ void EicPoint::mouseDoubleClickEvent(QGraphicsSceneMouseEvent*) {
     }
 
 }
-void EicPoint::mousePressEvent (QGraphicsSceneMouseEvent* event) {
-    //if (_group) _group->groupOveralMatrix();
 
+void EicPoint::mousePressEvent (QGraphicsSceneMouseEvent* event) {
     if (event->button() == Qt::RightButton)  {
         contextMenuEvent(event);
         return;
     }
 
     setZValue(10);
-
-    if (_group) Q_EMIT peakGroupSelected(_group);
-    if (_peak)  Q_EMIT peakSelected(_peak);
     if (_group)
-        _mw->groupRtWidget->plotGraph(_group);
-    if ( _group && _group->isIsotope() == false ) {
-        _mw->isotopeWidget->updateIsotopicBarplot(_group);
-    }
-
-    if(_scan) {
-        if (_mw->spectraDockWidget) {
-            _mw->spectraDockWidget->setVisible(true);
-            _mw->spectraDockWidget->raise();
-        }
-        if (_mw->spectraWidget->isVisible())
-            _mw->spectraWidget->setScan(_scan);
-            //_mw->peptideFragmentation->setScan(_scan);
-            if(_scan->mslevel >= 2)  _mw->spectralHitsDockWidget->limitPrecursorMz(_scan->precursorMz);
-
-    }
-
-    //if (_peak && _mw->spectraWidget->isVisible()) {
-    //    _mw->spectraWidget->setScan(_peak);
-    //}
-
-    if (_peak && _mw->covariantsPanel->isVisible()) {
-        _mw->getLinks(_peak);
-    }
-
-    if (_peak && _mw->adductWidget->isVisible()) {
-        _mw->adductWidget->setPeak(_peak);
-    }
-
-    if (_peak && _group && _mw->isotopeWidget->isVisible()) {
-        _mw->isotopeWidget->peakSelected(_peak, _group);
-    }
+        emit peakGroupSelected(_group);
+    if (_peak)
+        emit peakSelected(_peak);
+
+    // make changes through static functions, since this object might be
+    // destroyed during the execution period of those functions.
+    auto mw = &(*_mw);
+    auto group = &(*_group);
+    auto peak = &(*_peak);
+    auto scan = &(*_scan);
+    _updateWidgetsForPeakGroup(_mw, _group, _peak);
+    _updateWidgetsForScan(_mw, _scan);
```

Comment:
@saifulbkhan `mw`, `group`, `peak`, and `scan` are not being used, right?

Reply:
Ah! I forgot to remove this from an earlier implementation. The pointers will be copied as arguments. Will clear all the unnecessary copies.